dbms/src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -156,6 +157,18 @@ MergeTreeData::MergeTreeData(
         min_format_version = MERGE_TREE_DATA_MIN_FORMAT_VERSION_WITH_CUSTOM_PARTITIONING;
     }
 
+    for (const auto & [name, ast] : getColumns().ttl_expressions)
+    {
+        ASTPtr ttl_ast = ast;
+        auto syntax_result = SyntaxAnalyzer(global_context).analyze(ttl_ast, getColumns().getAllPhysical());
+        auto expr = ExpressionAnalyzer(ttl_ast, syntax_result, global_context).getActions(false);
```

Comment:
What if exception would be thrown here?
For example correct TTL expression but with absent column. I think we should catch exception, and throw it with error code `BAD_TTL_EXPRESSION`.